### ISSUE TITLE
Missing XOR support for booleans

### ISSUE THREAD

**Issue by filipw** [microsoft/qsharp#1096]
**Is your feature request related to a problem? Please describe.**
The old QDK had a handy [Xor](https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.logical.xor) function which could be used to perform XOR between two booleans. 

**Describe the solution you'd like**
It was a super simple function but very useful, so it could just be added back

```qsharp
    function Xor(a : Bool, b : Bool) : Bool {
        (a or b) and ((not a) or (not b))
    }
```

Alternatively the `^^^` (or `^`) could be extended to work with booleans - at the moment it works with ints only. For example in C# I could write `true ^ true` or `false ^ true`

**Describe alternatives you've considered**
1. do nothing and force everyone to write this by hand
2. add the function
3. extend the operator

**Additional context**
None

**Comment by swernli** [comment]
This is an interesting one... we could add the library function, but I'm wondering if actually makes sense to add a new keyword for this. I like the idea of having an operator, but if we allow `^^^` which is bitwise-xor to work on `Bool` types then for consistency it seems like it should work with the other bitwise operators (`&&&` and `|||`), which then become redundant with existing keywords `and` and `or`. In the AST, we already have support for `OrB` and `OrL` which correspond to bitwise OR `|||` and logical OR `or`, while there is a bitwise XOR `^^^` and no corresponding logical XOR `xor`.

Adding keywords to the language should be done with care and caution, so I don't propose it lightly. If were to take something that wasn't a keyword before and turn it into a keyword it can break existing code (if someone has a variable named `xor` suddenly their code would not compile). So having the function in the stdlib is the least disruptive option.

@sezna, @billti for FYI

**Comment by swernli** [comment]
After some discussion (that brought up some interesting thoughts regarding future cleanup of operators), it seems like our best options is to take the proposed function into the stdlib. This mirrors what some other languages have, where operators for Boolean logic don't include xor (support for using bitwise-xor on Boolean values usually comes from implicit type conversions, which Q# doesn't have).

@filipw, would you be willing to open a PR adding `Xor` to the standard library?

@DmitryVasilevsky and @cesarzc, this would be the first function in the `Microsoft.Quantum.Logical` namespace if we keep it in the same place, and it seems a little odd to me to have a namespace like that with only one function. Any suggestions of where else we might put it?

**Comment by msoeken** [comment]
XOR is the same as non-equality for Booleans, so you could simply use `a != b` for XOR.

**Comment by filipw** [comment]
> XOR is the same as non-equality for Booleans, so you could simply use `a != b` for XOR.

I personally think there is value in having explicit XOR operator support (or a dedicated function) for boolean values too. This helps with clarity, intent, readability and is especially helpful when more complex logical expressions arise, as well as in educational scenarios.

**Comment by msoeken** [comment]
> I personally think there is value in having explicit XOR operator support (or a dedicated function) for boolean values too. This helps with clarity, intent, readability and is especially helpful when more complex logical expressions arise, as well as in educational scenarios.

Agree, suggested that as an alternative to list for the issue and as a suggestion to simplify the implementation of the `Xor` function. Personally, I also prefer `a xor b` over `a != b` for clarity, but I prefer `a != b` over `Xor(a, b)` for readability.

**Comment by filipw** [comment]
Makes sense!
